org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -240,10 +239,12 @@ private String create() throws Exception {
 						//	Optional Restrictions
 						+ " AND (dl.Group1 IS NULL OR p.Group1=dl.Group1)"
 						+ " AND (dl.Group2 IS NULL OR p.Group2=dl.Group2)"
-						+ " AND (dl.C_BPartner_ID IS NULL OR EXISTS "
-						+ "(SELECT * FROM M_Product_PO po "
-						+ "WHERE po.M_Product_ID=p.M_Product_ID AND po.C_BPartner_ID=dl.C_BPartner_ID))"
-						+ " AND (dl.M_Product_ID IS NULL OR p.M_Product_ID=dl.M_Product_ID)");
+						+ " AND ((dl.C_BPartner_ID IS NULL AND dl.VendorCategory IS NULL) OR EXISTS "

```

Comment:
not convinced about the logic here.
what if a record just defines c_bpartner_id but not vendorcategory?
or the opposite - just vendor category and not business partner?

Maybe this comment from @hengsin makes sense:
-> I’m not sure VendorCategory is necessary when we already have > 1 way to group vendor.

Reply:
If only one of them is defined then only the defined one will select M_Product_PO records, because in the subselect the parts which are NULL are not evaluated.
AND (dl.C_BPartner_ID IS NULL OR po.C_BPartner_ID=dl.C_BPartner_ID)
AND (dl.VendorCategory IS NULL OR po.VendorCategory=dl.VendorCategory)))

Reply:
And actually there is no vendor specific grouping other than the vendor itself which at least for my customer isn't enough.

Reply:
I think we should keep the existing condition for "dl.C_BPartner_ID IS NULL OR EXISTS ..." and add another condition for dl.VendorCatory:
AND (dl.C_BPartner_ID IS NULL AND (dl.VendorCategory IS NULL OR EXISTS
(select * from m_product_po po where po.m_product_id=p.m_product_id and po.vendorcategory=dl.vendorcategory)))

As I've posted above, dl.C_BPartner_ID should take precedence over dl.VendorCategory

Reply:
If both are defined, both must match. Otherwise we would select product where by chance the vendorcategory is the same but the real product is totally different (vendor A uses category 47 for screws, vendor B uses 47 for shoes). And only if both are null there should be the OR alternative. Otherwise we would find Products which don't match the Vendorcategory.

Reply:
alright, change it to just 
AND (dl.VendorCategory IS NULL OR EXISTS
(select * from m_product_po po where po.m_product_id=p.m_product_id and po.vendorcategory=dl.vendorcategory))

Reply:
Just to be sure to have the correct idea of your suggested change: 
	 AND (dl.C_BPartner_ID IS NULL OR EXISTS 
	         (SELECT * FROM M_Product_PO po 
	            WHERE po.M_Product_ID=p.M_Product_ID AND po.C_BPartner_ID=dl.C_BPartner_ID))
	                AND (dl.M_Product_ID IS NULL OR p.M_Product_ID=dl.M_Product_ID)
         AND (dl.VendorCategory IS NULL OR EXISTS
                   (select * from m_product_po po where po.m_product_id=p.m_product_id 
                        and po.vendorcategory=dl.vendorcategory))

But this would select products where a po line exists with the correct vendor but a wrong vendorcategory and another po line with the correct vendorcategory but a wrong vendor. Of course this will happen very seldom but those errors are worst to find.